packages/messaging/ios/RNFBMessaging/RNFBMessaging+UNUserNotificationCenter.m

```diff
@@ -82,6 +83,50 @@ - (void)userNotificationCenter:(UNUserNotificationCenter *)center
        willPresentNotification:(UNNotification *)notification
          withCompletionHandler:
              (void (^)(UNNotificationPresentationOptions options))completionHandler {
+
+  NSArray *presentationOptionsConfig = [[RNFBJSON shared] getArrayValue:@"messaging_ios_foreground_presentation_options"
+                                                           defaultValue:@[]];
+
+  UNNotificationPresentationOptions presentationOptions = UNNotificationPresentationOptionNone;
+
+  BOOL badge = [presentationOptionsConfig containsObject:@"badge"];
+  BOOL sound = [presentationOptionsConfig containsObject:@"sound"];
+  BOOL alert = [presentationOptionsConfig containsObject:@"alert"];
+  BOOL list = [presentationOptionsConfig containsObject:@"list"];
+  BOOL banner = [presentationOptionsConfig containsObject:@"banner"];
+
+  if (badge) {
+    presentationOptions |= UNNotificationPresentationOptionBadge;
+  }
+
+  if (sound) {
+    presentationOptions |= UNNotificationPresentationOptionSound;
+  }
+
+  // if list or banner is true, ignore `alert` property
+  if (banner || list) {
+    if (banner) {
+      if (@available(iOS 14, *)) {
+        presentationOptions |= UNNotificationPresentationOptionBanner;
+      } else {
+        // for iOS 13 we need to set `alert`
+        presentationOptions |= UNNotificationPresentationOptionAlert;
+      }
+    }
+
+    if (list) {
+      if (@available(iOS 14, *)) {
+        presentationOptions |= UNNotificationPresentationOptionList;
+      } else {
+        // for iOS 13 we need to set `alert`
+        presentationOptions |= UNNotificationPresentationOptionAlert;
+      }
+    }
+  } else if (alert) {
+    // TODO: Remove `alert` once iOS 14 becomes the minimum deployment target
+    presentationOptions |= UNNotificationPresentationOptionAlert;
+  }
```

Comment:
Same logic in `notifee`

Reply:
yes - I was silently reading those as they went through, so I followed things as they went to the final solution, this looks good to me